qiskit/dagcircuit/dagcircuit.py

```diff
@@ -550,7 +548,7 @@ def compose(self, other, edge_map=None, front=False):
                 self._check_condition(nd.name, condition)
                 m_qargs = list(map(lambda x: edge_map.get(x, x), nd.qargs))
                 m_cargs = list(map(lambda x: edge_map.get(x, x), nd.cargs))
-                self.apply_operation_back(nd.op, m_qargs, m_cargs, condition)
+                self.apply_operation_back(nd.op, m_qargs, m_cargs)
```

Comment:
This looks like it is missing a step here to update `nd.op` (preferably a copy) with the output of `_map_condition`.

Reply:
> Ah! Thank you for catching this. Will make the changes, although could you please clarify the reason to use `nd.op.copy()` over `nd.op`.

Sure thing. In this case, since `nd` is from a user provided object (`other`), the copy is to prevent the modifications we make from affecting the user's original instance.
> 
> It's strange though that the tests didn't bring any errors related to the missing step. Is it viable to open an issue for more/ stricter testing here?

Agree, it looks like `dagcircuit/test_compose.py` doesn't have coverage for setting `condition`. An issue would be welcome!

Reply:
Thank you for the explanation! It appears that `test_compose.py` does have a `self.condition` variable, but it isn't used in any of the tests. Have opened an issue for it! Thanks for the go ahead.